tests/include/test/psa_exercise_key.h

```diff
@@ -242,15 +242,23 @@ int mbedtls_test_psa_exported_key_sanity_check(
  *
  * \param key       The key to exercise. It should be capable of performing
  *                  \p alg.
- * \param usage     The usage flags to assume.
- * \param alg       The algorithm to exercise.
+ * \param usage                    The usage flags to assume.
+ * \param alg                      The algorithm to exercise.
+ * \param expected_error_stage     Error stage while key exercise
+ *                                 to handle special cases.
+ *                                 0 - unused
+ *                                 1 - no error
+ *                                 2..n - error stages
+ * \param expected_status          Expected status at specified execution stage.
  *
  * \retval 0 The key failed the smoke tests.
  * \retval 1 The key passed the smoke tests.
  */
 int mbedtls_test_psa_exercise_key( mbedtls_svc_key_id_t key,
                                    psa_key_usage_t usage,
-                                   psa_algorithm_t alg );
+                                   psa_algorithm_t alg,
```

Comment:
I don't like adding all this complexity, just for the sake of a few unusual cases with key derivation. This would be better handled with a dedicated function for key derivation.